src/company_base.h

```diff
@@ -59,95 +59,87 @@ class FreeUnitIDGenerator {
 extern CompanyPool _company_pool;
 
 /** Statically loadable part of Company pool item */
 struct CompanyProperties {
-	uint32_t name_2;                   ///< Parameter of #name_1.
-	StringID name_1;                 ///< Name of the company if the user did not change it.
-	std::string name;                ///< Name of the company if the user changed it.
+	uint32_t name_2 = 0; ///< Parameter of #name_1.
+	StringID name_1 = INVALID_STRING_ID; ///< Name of the company if the user did not change it.
+	std::string name{}; ///< Name of the company if the user changed it.
 
-	StringID president_name_1;       ///< Name of the president if the user did not change it.
-	uint32_t president_name_2;         ///< Parameter of #president_name_1
-	std::string president_name;      ///< Name of the president if the user changed it.
+	StringID president_name_1 = INVALID_STRING_ID; ///< Name of the president if the user did not change it.
+	uint32_t president_name_2 = 0; ///< Parameter of #president_name_1
+	std::string president_name{}; ///< Name of the president if the user changed it.
 
-	NetworkAuthorizedKeys allow_list; ///< Public keys of clients that are allowed to join this company.
+	NetworkAuthorizedKeys allow_list{}; ///< Public keys of clients that are allowed to join this company.
 
-	CompanyManagerFace face;         ///< Face description of the president.
+	CompanyManagerFace face{}; ///< Face description of the president.
 
-	Money money;                     ///< Money owned by the company.
-	uint8_t money_fraction;             ///< Fraction of money of the company, too small to represent in #money.
-	Money current_loan;              ///< Amount of money borrowed from the bank.
-	Money max_loan;                  ///< Max allowed amount of the loan or COMPANY_MAX_LOAN_DEFAULT.
+	Money money = 0; ///< Money owned by the company.
+	uint8_t money_fraction = 0; ///< Fraction of money of the company, too small to represent in #money.
+	Money current_loan = 0; ///< Amount of money borrowed from the bank.
+	Money max_loan = COMPANY_MAX_LOAN_DEFAULT; ///< Max allowed amount of the loan or COMPANY_MAX_LOAN_DEFAULT.
 
-	Colours colour;                  ///< Company colour.
+	Colours colour = COLOUR_BEGIN; ///< Company colour.
 
-	uint8_t block_preview;              ///< Number of quarters that the company is not allowed to get new exclusive engine previews (see CompaniesGenStatistics).
+	uint8_t block_preview = 0; ///< Number of quarters that the company is not allowed to get new exclusive engine previews (see CompaniesGenStatistics).
 
-	TileIndex location_of_HQ;        ///< Northern tile of HQ; #INVALID_TILE when there is none.
-	TileIndex last_build_coordinate; ///< Coordinate of the last build thing by this company.
+	TileIndex location_of_HQ = INVALID_TILE; ///< Northern tile of HQ; #INVALID_TILE when there is none.
+	TileIndex last_build_coordinate{}; ///< Coordinate of the last build thing by this company.
```

Comment:
I know you already tried changing this, what prevents this being INVALID_TILE?

Also `... last built thing ...`

Reply:
It would require a save game conversion.

Changing this to `INVALID_TILE` means the code that sets the company name needs to be amended. Otherwise it'd try to find the town closest to `INVALID_TILE`.
That means that the current `0` is interpreted as something was built, which makes regression fail because the company has not built yet but the company name is changed to the town closest to tile `0`.
To prevent that a save game conversion is needed.

Should it also be `last_built_coordinate`?

Reply:
To me, `last_build_coordinate` is fine as a variable name, as it's the coordinate of the last build command.

`last_built_coordinate` would also be fine but of course needs things to change in the savegame code.

Yeah, English is funny :)